integration-tests/src/test/java/com/datastax/oss/driver/api/core/config/DriverConfigProfileReloadIT.java

```diff
@@ -0,0 +1,160 @@
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.config;
+
+import com.datastax.oss.driver.api.core.Cluster;
+import com.datastax.oss.driver.api.core.DriverTimeoutException;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.session.Session;
+import com.datastax.oss.driver.api.testinfra.simulacron.SimulacronRule;
+import com.datastax.oss.driver.categories.LongTests;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoader;
+import com.datastax.oss.simulacron.common.cluster.ClusterSpec;
+import com.typesafe.config.ConfigFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+
+import static com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoader.DEFAULT_CONFIG_SUPPLIER;
+import static com.datastax.oss.simulacron.common.stubbing.PrimeDsl.noRows;
+import static com.datastax.oss.simulacron.common.stubbing.PrimeDsl.when;
+import static org.junit.Assert.fail;
+
+@Category(LongTests.class)
+public class DriverConfigProfileReloadIT {
+
+  @Rule public SimulacronRule simulacron = new SimulacronRule(ClusterSpec.builder().withNodes(3));
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void should_periodically_reload_configuration() throws Exception {
+    String query = "mockquery";
+    // Define a loader which configures a reload interval of 2s and current value of configSource.
+    AtomicReference<String> configSource = new AtomicReference<>("");
+    DefaultDriverConfigLoader loader =
+        new DefaultDriverConfigLoader(
+            () ->
+                ConfigFactory.parseString("config-reload-interval = 2s\n" + configSource.get())
+                    .withFallback(DEFAULT_CONFIG_SUPPLIER.get()),
+            CoreDriverOption.values());
+    try (Cluster configCluster =
+        Cluster.builder()
+            .withConfigLoader(loader)
+            .addContactPoints(simulacron.getContactPoints())
+            .build()) {
+      simulacron.cluster().prime(when(query).then(noRows()).delay(2, TimeUnit.SECONDS));
+
+      Session session = configCluster.connect();
+
+      // Expect timeout since default timeout is .5 s
+      try {
+        session.execute(query);
+        fail("DriverTimeoutException expected");
+      } catch (DriverTimeoutException e) {
+        // expected.
+      }
+
+      // Bump up request timeout to 10 seconds and wait for config to reload.
+      configSource.set("request.timeout = 10s");
+      TimeUnit.SECONDS.sleep(3);
+
+      // Execute again, should not timeout.
+      session.execute(query);
+    }
+  }
+
+  @Test
+  public void should_not_allow_dynamically_adding_profile() throws Exception {
```

Comment:
It's a performance consideration: profiles are stored in an immutable map (`TpyeSafeDriverConfig#profiles`). If we allow new profiles at runtime, we have to turn it into a concurrent map.
The use case sounds extremely marginal to me, so I'm not sure if it's worth it.